File: 2022/10/main.py
cycles_of_interest = [20, 60, 100, 140, 180, 220]
signal_strength_sum = 0
x = 1
cycle = 0

with open("D:\\source\\repos\\advent-of-code\\2022\\10\\input.txt") as f:
    # Read the instructions from the file.
    instructions = f.readlines()

# Open the file containing the instructions.


    # Initialize the X register to 1 and the cycle number to 0.

    # Initialize a list of cycles of interest and a variable to store the sum of the
    # signal strengths.

    # Execute the instructions one by one until we reach a cycle of interest.
while len(cycles_of_interest) > 0:
        # Move to the next instruction.
    cycle += 1

        # Get the current instruction and its operand (if any).
    instruction, *operand = instructions[cycle % len(instructions)].split()

        # Execute the instruction.
    if instruction == "noop":
        pass
    elif instruction == "addx":
        x += int(operand[0])

        # If we have reached a cycle of interest, compute the signal strength and
        # add it to the running total.
    if cycle == cycles_of_interest[0]:
        signal_strength_sum += cycle * x
        cycles_of_interest.pop(0)

    # Output the sum of the signal strengths.
print(signal_strength_sum)

File: 2022/10/main.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
using namespace std;

int x = 1;
int cnt = 0;
int sm = 0;
vector<vector<char>> crt(6, vector<char>(40, ' '));

void cycle() {
  cnt += 1;
  if (cnt == 20 || cnt == 60 || cnt == 100 || cnt == 140 || cnt == 180 || cnt == 220) {
    sm += cnt * x;
  }
  if (abs((cnt - 1) % 40 - x) < 2) {
    crt[(cnt - 1) / 40][(cnt - 1) % 40] = '#';
  }
}

int main() {
  vector<string> ll;
  string line;
  ifstream input("input.txt");

  // Read input into ll
  while (getline(input, line)) {
    ll.push_back(line);
  }

  // Process instructions
  for (string line : ll) {
    if (line == "noop") {
      cycle();
    } else {
      int add = stoi(line.substr(5));
      cycle();
      cycle();
      x += add;
    }
  }

  // Print the result
  cout << sm << endl;
  for (auto line : crt) {
    for (char c : line) {
      cout << c;
    }
    cout << endl;
  }

  return 0;
}
